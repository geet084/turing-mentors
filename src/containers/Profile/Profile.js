import React, { Component } from 'react';
import { connect } from 'react-redux';

export class Profile extends Component {
  render() {
    
    return (
      <div>
       <h2>Profile page</h2>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  user: state.user,
})

export default connect(mapStateToProps)(Profile);