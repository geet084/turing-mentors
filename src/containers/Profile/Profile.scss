.profile {
  .Collapsible__contentInner {
    display: flex;
    flex-direction: column;
    p {
      margin: 5px;
    }
  }
}