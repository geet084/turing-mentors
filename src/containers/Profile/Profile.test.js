import React from 'react';
import { shallow } from 'enzyme';
import { Profile, mapStateToProps } from './Profile';

describe('Profile', () => {
  let wrapper;
  const mockProps = {
    user: {
      first_name: '',
      last_name: '',
      identities: [],
      location: '',
      cohort: '',
      program: '',
      current_job: '',
      slack: '',
      email: '',
      phone: '',
      linkedin: '',
      preferred_method: '',
      background: '',
      tech_skills: [],
      non_tech_skills: [],
      availability: {}
    }
  };

  beforeEach(() => {
    wrapper = shallow(<Profile {...mockProps} />);
  });

  describe('initial', () => {
    it('should have match the correct default snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should have match the correct props snapshot', () => {
      wrapper.setProps({
        user: {
          first_name: 'first',
          last_name: 'last',
          identities: [1],
          location: 'Denver',
          cohort: '1810',
          program: 'FE',
          current_job: 'Person',
          slack: 'slack',
          email: 'email',
          phone: '1910201231',
          linkedin: 'linkedin',
          preferred_method: '1',
          background: 'things and stuff',
          tech_skills: ['1', '3'],
          non_tech_skills: ['2', '4'],
          availability: {
            0: [false, false, true],
            1: [false, false, true],
            2: [false, false, true],
            3: [false, false, true],
            4: [false, false, true],
            5: [false, false, true],
            6: [false, false, true],
          }
        }
      })
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {
    it('should return a user object', () => {
      const mockState = {
        user: {},
        stuff: {},
        things: '',
      };

      const expected = {
        user: {}
      };

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });
});