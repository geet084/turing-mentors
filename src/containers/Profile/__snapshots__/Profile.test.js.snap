// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile initial should have match the correct default snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Profile
    user={
      Object {
        "availability": Object {},
        "background": "",
        "cohort": "",
        "current_job": "",
        "email": "",
        "first_name": "",
        "identities": Array [],
        "last_name": "",
        "linkedin": "",
        "location": "",
        "non_tech_skills": Array [],
        "phone": "",
        "preferred_method": "",
        "program": "",
        "slack": "",
        "tech_skills": Array [],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ProfileSection
          content={
            <UserInfo
              cohort=""
              current_job=""
              first_name=""
              identities={Array []}
              last_name=""
              profile={true}
              program=""
            />
          }
          name="User Info"
        />,
        <ProfileSection
          content={
            <UserBio
              email=""
              linkedin=""
              location=""
              phone=""
              preferred_method=""
              profile={true}
              slack=""
            />
          }
          name="Contact Info"
        />,
        <ProfileSection
          content={
            <UserBackground
              background=""
              profile={true}
            />
          }
          name="Background Info"
        />,
        <ProfileSection
          content={
            <UserSchedule
              availability={Object {}}
              profile={true}
            />
          }
          name="Availability Info"
        />,
        <ProfileSection
          content={
            <UserTechSkills
              profile={true}
              tech_skills={Array []}
            />
          }
          name="TechSkills"
        />,
        <ProfileSection
          content={
            <UserNonTechSkills
              non_tech_skills={Array []}
              profile={true}
            />
          }
          name="Non-TechSkills"
        />,
      ],
      "className": "link-content",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "content": <UserInfo
            cohort=""
            current_job=""
            first_name=""
            identities={Array []}
            last_name=""
            profile={true}
            program=""
          />,
          "name": "User Info",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "content": <UserBio
            email=""
            linkedin=""
            location=""
            phone=""
            preferred_method=""
            profile={true}
            slack=""
          />,
          "name": "Contact Info",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "content": <UserBackground
            background=""
            profile={true}
          />,
          "name": "Background Info",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "content": <UserSchedule
            availability={Object {}}
            profile={true}
          />,
          "name": "Availability Info",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "content": <UserTechSkills
            profile={true}
            tech_skills={Array []}
          />,
          "name": "TechSkills",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "content": <UserNonTechSkills
            non_tech_skills={Array []}
            profile={true}
          />,
          "name": "Non-TechSkills",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ProfileSection
            content={
              <UserInfo
                cohort=""
                current_job=""
                first_name=""
                identities={Array []}
                last_name=""
                profile={true}
                program=""
              />
            }
            name="User Info"
          />,
          <ProfileSection
            content={
              <UserBio
                email=""
                linkedin=""
                location=""
                phone=""
                preferred_method=""
                profile={true}
                slack=""
              />
            }
            name="Contact Info"
          />,
          <ProfileSection
            content={
              <UserBackground
                background=""
                profile={true}
              />
            }
            name="Background Info"
          />,
          <ProfileSection
            content={
              <UserSchedule
                availability={Object {}}
                profile={true}
              />
            }
            name="Availability Info"
          />,
          <ProfileSection
            content={
              <UserTechSkills
                profile={true}
                tech_skills={Array []}
              />
            }
            name="TechSkills"
          />,
          <ProfileSection
            content={
              <UserNonTechSkills
                non_tech_skills={Array []}
                profile={true}
              />
            }
            name="Non-TechSkills"
          />,
        ],
        "className": "link-content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "content": <UserInfo
              cohort=""
              current_job=""
              first_name=""
              identities={Array []}
              last_name=""
              profile={true}
              program=""
            />,
            "name": "User Info",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "content": <UserBio
              email=""
              linkedin=""
              location=""
              phone=""
              preferred_method=""
              profile={true}
              slack=""
            />,
            "name": "Contact Info",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "content": <UserBackground
              background=""
              profile={true}
            />,
            "name": "Background Info",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "content": <UserSchedule
              availability={Object {}}
              profile={true}
            />,
            "name": "Availability Info",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "content": <UserTechSkills
              profile={true}
              tech_skills={Array []}
            />,
            "name": "TechSkills",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "content": <UserNonTechSkills
              non_tech_skills={Array []}
              profile={true}
            />,
            "name": "Non-TechSkills",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Profile initial should have match the correct props snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Profile
    user={
      Object {
        "availability": Object {
          "0": Array [
            false,
            false,
            true,
          ],
          "1": Array [
            false,
            false,
            true,
          ],
          "2": Array [
            false,
            false,
            true,
          ],
          "3": Array [
            false,
            false,
            true,
          ],
          "4": Array [
            false,
            false,
            true,
          ],
          "5": Array [
            false,
            false,
            true,
          ],
          "6": Array [
            false,
            false,
            true,
          ],
        },
        "background": "things and stuff",
        "cohort": "1810",
        "current_job": "Person",
        "email": "email",
        "first_name": "first",
        "identities": Array [
          1,
        ],
        "last_name": "last",
        "linkedin": "linkedin",
        "location": "Denver",
        "non_tech_skills": Array [
          "2",
          "4",
        ],
        "phone": "1910201231",
        "preferred_method": "1",
        "program": "FE",
        "slack": "slack",
        "tech_skills": Array [
          "1",
          "3",
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ProfileSection
          content={
            <UserInfo
              cohort="1810"
              current_job="Person"
              first_name="first"
              identities={
                Array [
                  1,
                ]
              }
              last_name="last"
              profile={true}
              program="FE"
            />
          }
          name="User Info"
        />,
        <ProfileSection
          content={
            <UserBio
              email="email"
              linkedin="linkedin"
              location="Denver"
              phone="1910201231"
              preferred_method="1"
              profile={true}
              slack="slack"
            />
          }
          name="Contact Info"
        />,
        <ProfileSection
          content={
            <UserBackground
              background="things and stuff"
              profile={true}
            />
          }
          name="Background Info"
        />,
        <ProfileSection
          content={
            <UserSchedule
              availability={
                Object {
                  "0": Array [
                    false,
                    false,
                    true,
                  ],
                  "1": Array [
                    false,
                    false,
                    true,
                  ],
                  "2": Array [
                    false,
                    false,
                    true,
                  ],
                  "3": Array [
                    false,
                    false,
                    true,
                  ],
                  "4": Array [
                    false,
                    false,
                    true,
                  ],
                  "5": Array [
                    false,
                    false,
                    true,
                  ],
                  "6": Array [
                    false,
                    false,
                    true,
                  ],
                }
              }
              profile={true}
            />
          }
          name="Availability Info"
        />,
        <ProfileSection
          content={
            <UserTechSkills
              profile={true}
              tech_skills={
                Array [
                  "1",
                  "3",
                ]
              }
            />
          }
          name="TechSkills"
        />,
        <ProfileSection
          content={
            <UserNonTechSkills
              non_tech_skills={
                Array [
                  "2",
                  "4",
                ]
              }
              profile={true}
            />
          }
          name="Non-TechSkills"
        />,
      ],
      "className": "link-content",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "content": <UserInfo
            cohort="1810"
            current_job="Person"
            first_name="first"
            identities={
              Array [
                1,
              ]
            }
            last_name="last"
            profile={true}
            program="FE"
          />,
          "name": "User Info",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "content": <UserBio
            email="email"
            linkedin="linkedin"
            location="Denver"
            phone="1910201231"
            preferred_method="1"
            profile={true}
            slack="slack"
          />,
          "name": "Contact Info",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "content": <UserBackground
            background="things and stuff"
            profile={true}
          />,
          "name": "Background Info",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "content": <UserSchedule
            availability={
              Object {
                "0": Array [
                  false,
                  false,
                  true,
                ],
                "1": Array [
                  false,
                  false,
                  true,
                ],
                "2": Array [
                  false,
                  false,
                  true,
                ],
                "3": Array [
                  false,
                  false,
                  true,
                ],
                "4": Array [
                  false,
                  false,
                  true,
                ],
                "5": Array [
                  false,
                  false,
                  true,
                ],
                "6": Array [
                  false,
                  false,
                  true,
                ],
              }
            }
            profile={true}
          />,
          "name": "Availability Info",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "content": <UserTechSkills
            profile={true}
            tech_skills={
              Array [
                "1",
                "3",
              ]
            }
          />,
          "name": "TechSkills",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "content": <UserNonTechSkills
            non_tech_skills={
              Array [
                "2",
                "4",
              ]
            }
            profile={true}
          />,
          "name": "Non-TechSkills",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ProfileSection
            content={
              <UserInfo
                cohort="1810"
                current_job="Person"
                first_name="first"
                identities={
                  Array [
                    1,
                  ]
                }
                last_name="last"
                profile={true}
                program="FE"
              />
            }
            name="User Info"
          />,
          <ProfileSection
            content={
              <UserBio
                email="email"
                linkedin="linkedin"
                location="Denver"
                phone="1910201231"
                preferred_method="1"
                profile={true}
                slack="slack"
              />
            }
            name="Contact Info"
          />,
          <ProfileSection
            content={
              <UserBackground
                background="things and stuff"
                profile={true}
              />
            }
            name="Background Info"
          />,
          <ProfileSection
            content={
              <UserSchedule
                availability={
                  Object {
                    "0": Array [
                      false,
                      false,
                      true,
                    ],
                    "1": Array [
                      false,
                      false,
                      true,
                    ],
                    "2": Array [
                      false,
                      false,
                      true,
                    ],
                    "3": Array [
                      false,
                      false,
                      true,
                    ],
                    "4": Array [
                      false,
                      false,
                      true,
                    ],
                    "5": Array [
                      false,
                      false,
                      true,
                    ],
                    "6": Array [
                      false,
                      false,
                      true,
                    ],
                  }
                }
                profile={true}
              />
            }
            name="Availability Info"
          />,
          <ProfileSection
            content={
              <UserTechSkills
                profile={true}
                tech_skills={
                  Array [
                    "1",
                    "3",
                  ]
                }
              />
            }
            name="TechSkills"
          />,
          <ProfileSection
            content={
              <UserNonTechSkills
                non_tech_skills={
                  Array [
                    "2",
                    "4",
                  ]
                }
                profile={true}
              />
            }
            name="Non-TechSkills"
          />,
        ],
        "className": "link-content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "content": <UserInfo
              cohort="1810"
              current_job="Person"
              first_name="first"
              identities={
                Array [
                  1,
                ]
              }
              last_name="last"
              profile={true}
              program="FE"
            />,
            "name": "User Info",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "content": <UserBio
              email="email"
              linkedin="linkedin"
              location="Denver"
              phone="1910201231"
              preferred_method="1"
              profile={true}
              slack="slack"
            />,
            "name": "Contact Info",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "content": <UserBackground
              background="things and stuff"
              profile={true}
            />,
            "name": "Background Info",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "content": <UserSchedule
              availability={
                Object {
                  "0": Array [
                    false,
                    false,
                    true,
                  ],
                  "1": Array [
                    false,
                    false,
                    true,
                  ],
                  "2": Array [
                    false,
                    false,
                    true,
                  ],
                  "3": Array [
                    false,
                    false,
                    true,
                  ],
                  "4": Array [
                    false,
                    false,
                    true,
                  ],
                  "5": Array [
                    false,
                    false,
                    true,
                  ],
                  "6": Array [
                    false,
                    false,
                    true,
                  ],
                }
              }
              profile={true}
            />,
            "name": "Availability Info",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "content": <UserTechSkills
              profile={true}
              tech_skills={
                Array [
                  "1",
                  "3",
                ]
              }
            />,
            "name": "TechSkills",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "content": <UserNonTechSkills
              non_tech_skills={
                Array [
                  "2",
                  "4",
                ]
              }
              profile={true}
            />,
            "name": "Non-TechSkills",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
